matchbox_server/src/main.rs

```diff
@@ -1,25 +1,19 @@
 mod args;
-mod error;
-mod signaling;
+mod state;
+mod topology;
 
-use crate::signaling::{ws_handler, ServerState};
-use axum::{http::StatusCode, response::IntoResponse, routing::get, Router};
-use clap::Parser;
-use std::{net::SocketAddr, sync::Arc};
-use tower_http::{
-    cors::{Any, CorsLayer},
-    trace::{DefaultOnResponse, TraceLayer},
-    LatencyUnit,
+use crate::{
+    state::{RequestedRoom, RoomId, ServerState},
+    topology::MatchmakingDemoTopology,
 };
-use tracing::{info, Level};
-use tracing_subscriber::prelude::*;
-
-pub use args::Args;
-pub use matchbox_protocol::PeerId;
+use args::Args;
+use axum::{http::StatusCode, response::IntoResponse, routing::get};
+use clap::Parser;
+use matchbox_signaling::SignalingServerBuilder;
+use tracing::info;
 
-#[tokio::main]
-async fn main() {
-    // Initialize logger
+fn setup_logging() {
+    use tracing_subscriber::prelude::*;
```

Comment:
Why do we `use` inside the `fn`?

Reply:
My guess is that we used different logging crates between WASM/Native, I'll check out why or if I can put it in the top. Good catch.